geomesa-datastore-bundle/geomesa-datastore-processors/src/main/scala/org/geomesa/nifi/datastore/processor/AvroIngestProcessor.scala

```diff
@@ -11,30 +11,94 @@ package org.geomesa.nifi.datastore.processor
 
 import java.io.InputStream
 
+import org.apache.nifi.components.PropertyDescriptor
 import org.apache.nifi.flowfile.FlowFile
 import org.apache.nifi.processor._
 import org.apache.nifi.processor.io.InputStreamCallback
+import org.geomesa.nifi.datastore.processor.AbstractGeoIngestProcessor.checkCompatibleSchema
+import org.geomesa.nifi.datastore.processor.AvroIngestProcessor.{AvroMatchMode, buildWriter}
 import org.geotools.data._
+import org.geotools.feature.simple.SimpleFeatureBuilder
+import org.geotools.util.Converters
 import org.locationtech.geomesa.features.avro.AvroDataFileReader
 import org.locationtech.geomesa.utils.geotools._
-import org.opengis.feature.simple.SimpleFeatureType
+import org.opengis.feature.simple.{SimpleFeature, SimpleFeatureType}
 
+import scala.util.{Failure, Success}
 import scala.util.control.NonFatal
 
+object AvroIngestProcessor {
+  val AvroMatchMode: PropertyDescriptor =
+    new PropertyDescriptor.Builder()
+      .name("Avro SFT match mode")
+      .description("Determines how Avro SFT mismatches are handled")
+      .required(false)
+      .defaultValue("strict")
+      .allowableValues("strict", "lenient")
+      .build()
+
+  def buildWriter(sft1: SimpleFeatureType,
+                  sft2: SimpleFeatureType,
+                  matchMode: String): (FeatureWriter[SimpleFeatureType, SimpleFeature], SimpleFeature) => Unit = {
+    checkCompatibleSchema(sft1, sft2) match {
+      case Success(true) =>
+        // Schemas match, so use a regular writer.
+        // TODO:  Discuss if the useProvidedFid should be true!
```

Comment:
i guess this PR actually changes the behavior from false to true? that probably makes sense to do.

Reply:
Yes, this PR changes.  The TODO is catch our attention and make sure we are good with that decision.;)

Reply:
we could always make it a property. i'm not sure if anyone is relying on the current behavior or not, but it seems like something that could mess someone up either way.

Reply:
Indeed!  Yeah, that may be worth a property.  Let me add that in this PR.